staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -654,7 +664,7 @@ func NewDiscoveryClientForConfigAndClient(c *restclient.Config, httpClient *http
 		return nil, err
 	}
 	client, err := restclient.UnversionedRESTClientForConfigAndClient(&config, httpClient)
-	return &DiscoveryClient{restClient: client, LegacyPrefix: "/api"}, err
+	return &DiscoveryClient{restClient: client, LegacyPrefix: "/api", UseLegacyDiscovery: false}, err
```

Comment:
How do you expect discovery users to set that flag to true?

Reply:
Please look at PR description

Reply:
Thanks Sean, that's the answer I was looking for!

Reply:
What about other controllers in case we have a problem with them?

Reply:
If I use kubernetes.NewForClient().Discovery(), how would I set this?

Reply:
The result of that call would have to be a cast to `DiscoveryClient` (since it's returning a `DiscoveryInterface`), then the field set.

Reply:
If the returned discovery client is always a `DiscoveryClient`, can we return that (and avoid the upcast in the caller)? Should we have a new parameter? Should we have a new function? I think we have a lot of options that are cleaner than forcing an upcast from the caller.

Reply:
> If the returned discovery client is always a DiscoveryClient, can we return that (and avoid the upcast in the caller)?

-1 on changing the return interface to a concrete type.  Adding an interface function or a new function might be ok.  Ultimately this looks a lot like trying to have a featuregate without having a featuregate.

I know the client-side disabling even if the server exposes the endpoint is important to Daniel, so tagging him in for a preference.

/cc @lavalamp

Reply:
Can you just add a `WithLegacyDiscovery() DiscoveryClient` function to the interface?